UIKitTests/UIScrollViewTests.swift

```diff
@@ -47,6 +47,42 @@ class UIScrollViewTests: XCTestCase {
         XCTAssertEqual(scrollView.contentInset, arbitraryContentInset)
     }
 
+    func testIndicatorsPositionInHierarchyWhenAddingSubviews() {
+        // subviews[0] is the back-most view, higher indexes are higher in layer hierarchy
+        // we want scroll indicators to always remain on top of 'normal' subviews
+        // and we want newer 'normal' subviews to be above older ones
+
+        // XXX: iOS does not allow explicit access to scroll indicators via subviews.
+        // We should remove that, too - but for now it's the easy way to go and helpful in testing
+
+        let mockView = UIView()
+        scrollView.addSubview(mockView)
+
+        XCTAssertGreaterThan(
+            scrollView.subviews.index(of: scrollView.horizontalScrollIndicator)!,
```

Comment:
These tests look seriously amazing now. Great work @janek!

Reply:
that's too nice! thanks a lot!!